### DISCUSSION TITLE
[Feature]: Make path separator configurable

### DISCUSSION THREAD

**Discussion by MasouShizuka** [olimorris/persisted.nvim#97] [Ideas]
Currently `persisted.nvim` maps both `:` and `/` in the working directory to `%`, while `git branch separator` is configurable.
There will be some trouble if want to map the session file back to the working directory.
For example, on Windows, since the path contains both `:` and `/`, the reverse mapping of `%` is somewhat troublesome.

It is beneficial to make the session file mappable back to the working directory.
For example, I can use other launcher software to start neovim in the specific working directory based on the saved session file, so that `persisted.nvim` can `autoload` the session and achieve the startup function of workspace.

**Comment by olimorris** [comment]
It actually is. I just don't document it as.

**Comment by MasouShizuka** [comment]
I mean configure for `:` and `/` respectively.

Currently `:` and `/` are identically replaced by `%`.
```lua
function M.get_dir_pattern()
  local pattern = "/"
  if vim.fn.has("win32") == 1 then
    pattern = "[\\:]"
  end
  return pattern
end

local function get_current()
  local name = vim.fn.getcwd():gsub(utils.get_dir_pattern(), "%%")
  return config.options.save_dir .. name .. M.get_branch() .. ".vim"
end
```

**Reply by olimorris** [reply]
Do you have an example use of case of when you'd want to do this?

**Reply by MasouShizuka** [reply]
First of all, this feature can customize the name of saved session files, although I don't really care about it.

For the use of case, I wrote a plugin for the launcher program to scan `save_dir` and list all the saved session files. Then it needs to map the working directory of the session file based on the file name.
If I can get the corresponding working directory, I can let it start neovim in that location, and then `persisted.nvim` will autoread the saved session.

Then, for example, there is such a saved session file on Windows:
```
D%%Projects%python%hello@@main.vim
```
Since both `:` and `/` are mapped to `%`, I need to specifically determine whether `:` exists.

And if the mapping names of `:` and `/` are different, such as `:` -> `++`, `/` -> `__`, like [neovim-session-manager](https://github.com/Shatur/neovim-session-manager/blob/68dde355a4304d83b40cf073f53915604bdd8e70/lua/session_manager/config.lua#L4-L5), then I only need to replace `++` and `__` back to the corresponding path separator , without the need to specifically determine whether this path should contain `:`.

To put it simply, this actually makes it easier to map back to the working directory based on the session name.

**Reply by MasouShizuka** [reply]
By the way, if `use_git_branch=false`, maybe the `@@main` part should not be added.

**Reply by olimorris** [reply]
Gotcha! I'm open to a PR for this but I'd be wary of breaking functionality for existing Windows users and I'm not sure there's a clean way of implementing what you want without doing that.
